### PR TITLE
Prevent customer creation on backend for invalid wordpress roles with unit tests

### PR DESCRIPTION
**Pull request by jnmrclmbsjse**
This PR includes:
1. Only allow `subscriber` and `customer` roles to be synched as customers on backend.
2. Unit tests for user creation to customer synching.

### CODE REVIEW

File: tests/unit/FileExports/CustomerFileExportTest.php [hunk 1]
```diff
@@ -14,6 +17,35 @@ public function getFileExportClass(): string
         return CustomerFileExport::class;
     }
 
+    public function testUserCreateByRoles()
```
Comment (ikulis) [inline tests/unit/FileExports/CustomerFileExportTest.php:20]: it's better to use a data provider for the roles

**Review by ikulis — COMMENTED** [review]
one improvement

File: tests/unit/FileExports/CustomerFileExportTest.php (hunk 1 above)
Comment (jnmrclmbsjse) [reply tests/unit/FileExports/CustomerFileExportTest.php]: added data provier for test

File: tests/unit/FileExports/CustomerFileExportTest.php [hunk 2]
```diff
@@ -14,6 +17,51 @@ public function getFileExportClass(): string
         return CustomerFileExport::class;
     }
 
+    public function dataProviderUserCreateByRoles()
+    {
+        $tests = [];
+        foreach (GOCustomer::VALID_ROLES as $role) {
+            $tests['valid-'.$role] = [$role, $role.'_test'];
```
Comment (ikulis) [inline tests/unit/FileExports/CustomerFileExportTest.php:24]: you can just put  `_test` inside the test
```suggestion
            $tests['valid-'.$role] = [$role];
```

File: tests/unit/FileExports/CustomerFileExportTest.php [hunk 3]
```diff
@@ -14,6 +17,51 @@ public function getFileExportClass(): string
...
+        }
+        foreach (GOCustomer::INVALID_ROLES as $role) {
+            $tests['invalid-'.$role] = [$role, $role.'_test'];
```
Comment (ikulis) [inline tests/unit/FileExports/CustomerFileExportTest.php:27]: You can make the second parameter is you expect it to be valid or not, the current assertions are hard to understand
```suggestion
            $tests['invalid-'.$role] = [$role, false];
```

File: tests/unit/FileExports/CustomerFileExportTest.php [hunk 4]
```diff
@@ -14,6 +17,51 @@ public function getFileExportClass(): string
...
+        wp_set_current_user(1);
+        StoreKeeperApi::$mockAdapter->withModule(
+            'ShopModule',
+            function (MockInterface $module) {
+                $module->shouldReceive('newShopCustomer')->andReturnUsing(
+                    function ($got) {
+                        $user = $got[0]['relation'];
+                        // Passed firstname as the role in test data
+                        $role = $user['contact_person']['firstname'];
+                        $this->assertNotContains($role, GOCustomer::INVALID_ROLES, $role.' should not be returned');
```
Comment (ikulis) [inline tests/unit/FileExports/CustomerFileExportTest.php:49]: this functions is not beeing called then you make the invalid role how does this asset work than?

**Pull request merged**